src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -112,7 +124,7 @@ static void purgeExecutors() {
      * @return true if the policy was successfully enabled 
      */
     public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
-        if (!PURGE_FORCE) {
+        if (!PURGE_FORCE && (!IS_ANDROID || IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)) {
```

Comment:
Maybe I'm not understanding something because it's 10 pm and I'm still in the office... 

But your code will always return `false` on regular JVM because `IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY` will be `false`.